dataset/src/main/java/org/keycloak/benchmark/dataset/organization/OrganizationIdentityProviderProvisioner.java

```diff
@@ -0,0 +1,144 @@
+    }
+
+    @Path("create")
+    @GET
+    @NoCache
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response create() {
+        return start("Creation of " + config.getCount() + " identity providers in organization " + organization.getName(), addIdentityProviders());
+    }
+
+    private Runnable addIdentityProviders() {
+        return () -> {
+            String orgId = organization.getId();
+            CountDownLatch latch = new CountDownLatch(config.getCount());
+            KeycloakSessionFactory sessionFactory = baseSession.getKeycloakSessionFactory();
+            ExecutorHelper executor = new ExecutorHelper(config.getThreadsCount(), sessionFactory, config);
+
+            executor.addTask(() -> runJobInTransactionWithTimeout(session -> addIdentityProviders(session, orgId, latch)));
+
+            try {
+                executor.waitForAllToFinish();
+                success();
+                logger.infof("Added %d identity providers to organization %s", config.getCount() - latch.getCount(), organization.getName());
+            } catch (Exception e) {
+                KeycloakModelUtils.runJobInTransaction(sessionFactory, session
+                        -> new TaskManager(session).removeExistingTask(false));
+            }
+        };
+    }
+
+    protected void addIdentityProviders(KeycloakSession session, String orgId, CountDownLatch latch) {
+        OrganizationProvider provider = getOrganizationProvider(session);
+        OrganizationModel organization = provider.getById(orgId);
+        int startIndex = getLastIndex(session, organization.getName());
+        RealmModel realm = session.getContext().getRealm();
+        ClientModel client = realm.getClientByClientId("org-broker-client");
+
+        if (client == null) {
+            client = realm.addClient("org-broker-client");
+            client.setSecret("secret");
+            client.setProtocol(OIDCLoginProtocol.LOGIN_PROTOCOL);
+            client.setPublicClient(false);
+        }
+
+        long count = latch.getCount();
+
+        for (int i = startIndex; i < count; i++) {
+            String idpAlias = "idp-" + organization.getName() + "-" + i;
+
+            IdentityProviderModel identityProvider = realm.getIdentityProviderByAlias(idpAlias);
+
+            if (identityProvider != null && identityProvider.getInternalId() != null && identityProvider.getOrganizationId() != null) {
+                continue;
+            }
+
+            if (identityProvider == null || identityProvider.getInternalId() == null) {
+                identityProvider = new IdentityProviderModel();
+
+                identityProvider.setAlias(idpAlias);
+                identityProvider.setProviderId(KeycloakOIDCIdentityProviderFactory.PROVIDER_ID);
+                identityProvider.setLoginHint(true);
+                identityProvider.setEnabled(true);
+                HashMap<String, String> idpConfig = new HashMap<>();
+                identityProvider.setConfig(idpConfig);
```

Comment:
Do we want to have `idpConfig.put("kc.org.broker.public", "true")` configurable here?

Reply:
Yeah. Not adding now because not yet needed for the work I'm doing.